### PR TITLE
Allow partial `TorchToTcp` conversions using a whitelist

### PR DESCRIPTION
**Pull request by sjain-stanford**
Add pass option to TorchToTcp accepting a set of op name strings as whitelist (for selective conversion). This allows flexibility with phase ordering of conversion passes in downstream pipelines.

### CODE REVIEW

**Review by zezhang — APPROVED** [review]
Thanks for enabling this feature, LGTM!

File: lib/Conversion/TorchToTcp/TorchToTcp.cpp [hunk 1]
```diff
@@ -49,6 +60,15 @@ class ConvertTorchToTcp : public ConvertTorchToTcpBase<ConvertTorchToTcp> {
 
   void runOnOperation() override {
     MLIRContext *context = &getContext();
+    RewritePatternSet patterns(context);
+
+    // The strings in the `convertTorchOps` ArrayRef don't exist during the call
```
Comment (sanjoy) [inline lib/Conversion/TorchToTcp/TorchToTcp.cpp:65]: `The strings in the `convertTorchOps` ArrayRef don't exist` is weirdly phrased IMO; is it correct to say that usually the default constructor is called which means `convertTorchOps` is usually unset.

File: lib/Pipeline/Pipeline.cpp [hunk 2]
```diff
@@ -35,7 +35,10 @@
 using namespace mlir;
 
 static void createTorchBackendToTcpBackendPipeline(OpPassManager &pm) {
-  pm.addNestedPass<func::FuncOp>(tcp::createConvertTorchToTcpPass());
+  ArrayRef<std::string> emptyArrayRef;
+
+  pm.addNestedPass<func::FuncOp>(
+      tcp::createConvertTorchToTcpPass(emptyArrayRef));
```
Comment (sanjoy) [inline lib/Pipeline/Pipeline.cpp:41]: Can this be `tcp::createConvertTorchToTcpPass({})`?

**Review by sanjoy — APPROVED** [review]

File: lib/Conversion/TorchToTcp/Elementwise.cpp [hunk 3]
```diff
@@ -650,78 +650,59 @@ class ConvertAtenToDtypeOp : public OpConversionPattern<AtenToDtypeOp> {
...
-      typeConverter, context);
-
-  target.addIllegalOp<AtenBatchNormOp>();
-  patterns.add<ConvertAtenBatchNormOp>(typeConverter, context);
-
-  target.addIllegalOp<AtenAtan2Op>();
-  patterns.add<ConvertAtenAtan2Op>(typeConverter, context);
+    ConversionTarget &target, const llvm::StringSet<> &convertTorchOpsSet) {
+
+#define INSERT_ATEN_ELEMENTWISE_OP_PATTERN(ConvertAtenOpPattern, AtenOp)       \
```
[the remark this answers is not in the document]
Comment (sjain-stanford) [reply lib/Conversion/TorchToTcp/Elementwise.cpp]: Good suggestion, done. I think this was applicable to just `INSERT_ATEN_ELEMENTWISE_OP_PATTERN` macro in Elementwise.cpp and `INSERT_ATEN_MISC_OP_PATTERN` macro in Misc.cpp. PLMK if I missed any other occurrence and I'll address in a follow-on.

File: lib/Conversion/TorchToTcp/TorchToTcp.cpp (hunk 1 above)
Comment (sjain-stanford) [reply lib/Conversion/TorchToTcp/TorchToTcp.cpp]: Rephrased as suggested.

File: lib/Pipeline/Pipeline.cpp (hunk 2 above)
Comment (sjain-stanford) [reply lib/Pipeline/Pipeline.cpp]: Done.

**Pull request merged**